src/esockd_acceptor.erl

```diff
@@ -171,48 +202,49 @@ handle_event(
 ) ->
     %% make it look like gen_tcp:accept
     inet_db:register_socket(Sock, SockMod),
-
-    %% Inc accepted stats.
-    esockd_server:inc_stats({Proto, ListenOn}, accepted, 1),
-
     case eval_tune_socket_fun(TuneFun, Sock) of
-        {ok, Sock} ->
-            case esockd_connection_sup:start_connection(ConnSup, Sock, UpgradeFuns) of
+        {ok, NewSock} ->
```

Comment:
why the socket is changed?

Reply:
the socket does not change most of the time, but the API is designed to allow it to change.